### PR TITLE
Implemented mod()

### PR DESCRIPTION
**Pull request by TheSlimvReal**
Targets issue #124.

### CODE REVIEW

File: heat/core/arithmetics.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
     'div',
     'mul',
     'pow',
-    'sub'
+    'sub',
+    'mod'
```
Comment (Markus-Goetz) [inline heat/core/arithmetics.py]: Please sort this alphabetically

File: heat/core/tests/test_arithmetics.py [hunk 2]
```diff
@@ -125,6 +124,47 @@ def test_div(self):
         with self.assertRaises(TypeError):
             ht.div('T', 's')
 
+    def test_mod(self):
```
Comment (Markus-Goetz) [inline heat/core/tests/test_arithmetics.py]: Please provide a test case with a non-zero divisor (e.g. a % 2.4)

**Comment by coquelin77** [comment]
should this be 'fmod' instead of 'mod'? I dont see a huge difference between the two but i think we should stick with the numpy naming scheme. also from their docs:

"This is the NumPy implementation of the C library function fmod, the remainder has the same sign as the dividend x1. It is equivalent to the Matlab(TM) rem function and should not be confused with the Python modulus operator x1 % x2."

**Comment by TheSlimvReal** [comment]
> should this be 'fmod' instead of 'mod'? I dont see a huge difference between the two but i think we should stick with the numpy naming scheme. also from their docs:
> 
> "This is the NumPy implementation of the C library function fmod, the remainder has the same sign as the dividend x1. It is equivalent to the Matlab(TM) rem function and should not be confused with the Python modulus operator x1 % x2."
> 
> i guess my point is that i believe that if we implement a wrapper for a torch function in heat it should have the same name as the torch function

I understand the naming point. But would you then suggest to remove the __mod__ implementation because torch only offers the fmod functionality?

**Comment by Markus-Goetz** [comment]
For all intents and purposes mod() and fmod() are currently aliases. fmod() converts the input tensor beforehand to a double-precision floating-point number, whereas mod not necessarily does that. Since we do not have an alternative implementation for mod() both are the same.

**Pull request merged**